openwisp_notifications/static/openwisp-notifications/js/notifications.js

```diff
@@ -11,9 +12,56 @@ if (typeof gettext === 'undefined') {
         notificationWidget($);
         initNotificationDropDown($);
         initWebSockets($);
+        addToWindowList();
     });
 })(django.jQuery);
 
+window.addEventListener("beforeunload", function () {
+    removeFromWindowList();
+});
+
+function addToWindowList() {
+    // add window id to 'windows' in localStorage
+    windowId = String(Date.now());
+    let tempWindowsList = getWindowList();
+    let index = tempWindowsList.indexOf(windowId);
+    if (index < 0){
+        tempWindowsList.push(windowId);
+    }
+    localStorage.windows = tempWindowsList.join(',');
```

Comment:
please use `localStorage.notificationWindows`, which is more specific and easy to debug